### PR TITLE
Make GenFacades errors appear in MSbuild logs

### PR DESCRIPTION
**Pull request by danmosemsft**
Relates to https://github.com/dotnet/corefx/pull/29813

### CODE REVIEW

**Comment by danmosemsft** [comment]
eg
```
FillPartialFacadeUsingTask:
  Moving file from "C:\git\corefx\bin/obj/Windows_NT.AnyCPU.Debug/mscorlib/netcoreapp/mscorlib.dll" to "C:\git\corefx\bin/obj/Windows_NT.AnyCPU.Debug/mscorlib/netcoreapp/PreGenFacades/mscorlib.dll".
  Moving file from "C:\git\corefx\bin/obj/Windows_NT.AnyCPU.Debug/mscorlib/netcoreapp/mscorlib.pdb" to "C:\git\corefx\bin/obj/Windows_NT.AnyCPU.Debug/mscorlib/netcoreapp/PreGenFacades/mscorlib.pdb".
C:\git\corefx\Tools\partialfacades.task.targets(64,5): error : The type 'T:System.Collections.Concurrent.ConcurrentQueue`1' is defined in multiple seed assemblies. If this is intentional, specify one of the following arguments to c
hoose the preferred seed type:  /preferSeedType:System.Collections.Concurrent.ConcurrentQueue`1=System.Collections.Concurrent  /preferSeedType:System.Collections.Concurrent.ConcurrentQueue`1=System.Private.CoreLib [C:\git\corefx\sr
c\shims\manual\mscorlib.csproj]
C:\git\corefx\Tools\partialfacades.task.targets(64,5): error : The type 'T:System.Collections.Concurrent.IProducerConsumerCollection`1' is defined in multiple seed assemblies. If this is intentional, specify one of the following ar
guments to choose the preferred seed type:  /preferSeedType:System.Collections.Concurrent.IProducerConsumerCollection`1=System.Collections.Concurrent  /preferSeedType:System.Collections.Concurrent.IProducerConsumerCollection`1=Syst
em.Private.CoreLib [C:\git\corefx\src\shims\manual\mscorlib.csproj]
C:\git\corefx\Tools\partialfacades.task.targets(64,5): error : Errors were encountered while generating the facade. [C:\git\corefx\src\shims\manual\mscorlib.csproj]
C:\git\corefx\Tools\partialfacades.task.targets(64,5): error : Errors were encountered when generating facade(s). [C:\git\corefx\src\shims\manual\mscorlib.csproj]
Done Building Project "C:\git\corefx\src\shims\manual\mscorlib.csproj" (default targets) -- FAILED.
```

File: src/Microsoft.DotNet.Build.Tasks/GenFacadesTask.cs [hunk 1]
```diff
@@ -125,6 +125,29 @@ public TraceLogger(TaskLoggingHelper log)
...
+            else if (eventType == TraceEventType.Warning)
+            {
+                _log.LogWarning(message);
+            }
+            else
+            {
+                _log.LogMessage(message);
+            }
+        }        
+
```
Comment (danmosemsft) [inline src/Microsoft.DotNet.Build.Tasks/GenFacadesTask.cs]: Left the Write override as Trace.Log still goes through here.

File: src/GenFacades/GenFacades.Core/GenFacades.cs [hunk 2]
```diff
@@ -637,12 +637,15 @@ private INamedTypeDefinition GetSeedType(string docId, IReadOnlyList<INamedTypeD
...
+                var sb = new StringBuilder();
+                sb.AppendFormat("The type '{0}' is defined in multiple seed assemblies. If this is intentional, specify one of the following arguments to choose the preferred seed type:", docId);
 
                 foreach (INamedTypeDefinition type in seedTypes)
                 {
-                    Trace.TraceError("  /preferSeedType:{0}={1}", docId.Substring("T:".Length), type.GetAssembly().Name.Value);
+                    sb.AppendFormat("  /preferSeedType:{0}={1}", docId.Substring("T:".Length), type.GetAssembly().Name.Value);
                 }
+
+                Trace.TraceError(sb.ToString());
```
Comment (danmosemsft) [inline src/GenFacades/GenFacades.Core/GenFacades.cs]: Create a single error so it's not interleaved in log

**Comment by danmosemsft** [comment]
@weshaggard

File: src/Microsoft.DotNet.Build.Tasks/GenFacadesTask.cs [hunk 3]
```diff
@@ -125,6 +125,29 @@ public TraceLogger(TaskLoggingHelper log)
...
+            {
+                _log.LogError(message);
+            }
+            else if (eventType == TraceEventType.Warning)
+            {
+                _log.LogWarning(message);
+            }
+            else
+            {
+                _log.LogMessage(message);
```
[the remark this answers is not in the document]
Comment (danmosemsft) [reply src/Microsoft.DotNet.Build.Tasks/GenFacadesTask.cs]: I don't know enough about the message content to know whether we want it to show up in normal verbosity or not (which most folks get if they type msbuild). So I left at normal. Happy to reduce if it's noisy.

[the remark this answers is not in the document]
Comment (weshaggard) [reply src/Microsoft.DotNet.Build.Tasks/GenFacadesTask.cs]: I suspect it was already doing that before so keeping it the same is probably OK.

**Pull request merged**